core/chains/evm/gas/models.go

```diff
@@ -34,6 +34,9 @@ type EvmFeeEstimator interface {
 
 	// L1Oracle returns the L1 gas price oracle only if the chain has one, e.g. OP stack L2s and Arbitrum.
 	L1Oracle() rollups.L1Oracle
+	// GetConfig returns the underlying gas estimator config
+	GetConfig() evmconfig.GasEstimator
```

Comment:
I think this approach is too heavy handed because it inverts the relationship and forces every estimator to use the same unified configuration, including fields that are not relevant to it.

Is `PriceMax` the only field used?